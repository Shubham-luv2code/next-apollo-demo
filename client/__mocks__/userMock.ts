export const userMock = [
  {
    uuid: "fef7ef45-9aaf-4e8c-bd9a-442ed8b7832d",
    fullName: "Stanton Goodwin",
    address: "586 Madonna Plain\nWest Christian, SC 90789",
    city: "West Jaquelin",
    country: "Ireland",
    phoneNumber: "549-842-0265",
    email: "Carol_Zboncak@Trycia.co.uk",
    __typename: "User",
  },
  {
    uuid: "f90aa7ba-729c-4ab2-8c48-8d7d9cdb4273",
    fullName: "Juana Douglas",
    address: "7313 Hannah Key\nWilhelminemouth, WA 41038-1578",
    city: "Elbertfort",
    country: "Malaysia",
    phoneNumber: "533-257-6954",
    email: "Swift.Lisette@gmail.com",
    __typename: "User",
  },
  {
    uuid: "bf815158-179b-4241-af54-11c27b20253b",
    fullName: "Marcus Ankunding",
    address: "266 White Squares\nLake Araside, NJ 19408",
    city: "New Jacintheview",
    country: "Nepal",
    phoneNumber: "197-409-9973",
    email: "Douglas_Elmore@gmail.com",
    __typename: "User",
  },
  {
    uuid: "709e59b4-b146-434e-ba3a-a9b45809c8b8",
    fullName: "Skye Balistreri",
    address: "517 Tiara Islands Apt. 318\nSouth Karlie, AL 57802-2337",
    city: "New Robbie",
    country: "Ireland",
    phoneNumber: "183-762-1794",
    email: "Erika.Klein@gmail.com",
    __typename: "User",
  },
  {
    uuid: "eb225516-0c7f-4b18-be8e-a2bd6de39fa2",
    fullName: "Billy Stamm",
    address: "136 Stracke Alley\nLake Winston, SD 43995-8763",
    city: "Imeldastad",
    country: "Suriname",
    phoneNumber: "899-207-8259",
    email: "Jonas_Kovacek@Dino.tv",
    __typename: "User",
  },
  {
    uuid: "4953922c-a390-4e03-b257-e1cfabe7a31f",
    fullName: "Dayne Douglas",
    address: "291 Anderson Ports Apt. 845\nLake Jerald, CO 80425",
    city: "South Estebanfurt",
    country: "Marshall Islands",
    phoneNumber: "237-292-0511",
    email: "Sven.Sauer@Makenzie.us",
    __typename: "User",
  },
  {
    uuid: "e8bfeee6-dbf1-46a4-a7c0-071360f1d795",
    fullName: "Rosalinda Stehr",
    address: "07 Rice Lakes\nWest Dennis, CA 91308",
    city: "Lake Allan",
    country: "Poland",
    phoneNumber: "522-978-5728",
    email: "Neha.Mante@Olaf.biz",
    __typename: "User",
  },
  {
    uuid: "d2a62aa3-4052-4a08-919a-9b65ab33c8dc",
    fullName: "Owen Tillman",
    address: "939 Jerde Route Suite 332\nEast Melba, CT 99777",
    city: "South Martina",
    country: "Ukraine",
    phoneNumber: "573-484-5973",
    email: "Paucek_Jedidiah@hotmail.com",
    __typename: "User",
  },
  {
    uuid: "a981efce-a800-4ef2-a4de-07f0b859ab1d",
    fullName: "Henry Reilly",
    address: "975 Reinger Walks\nOrphaberg, LA 81545-5765",
    city: "Kossberg",
    country: "Svalbard & Jan Mayen Islands",
    phoneNumber: "587-853-2000",
    email: "Hannah.Zboncak@gmail.com",
    __typename: "User",
  },
  {
    uuid: "bd438595-2288-4bc6-9cda-1940de5643fc",
    fullName: "Osborne Cummings",
    address: "7748 Adah Loaf\nSouth Dellaland, MD 29907-1675",
    city: "Charlottehaven",
    country: "Malta",
    phoneNumber: "978-462-6869",
    email: "Hal.McCullough@Orn.info",
    __typename: "User",
  },
  {
    uuid: "d940e6c3-7122-4ae5-a845-e7a5c1583a59",
    fullName: "Callie Ritchie",
    address: "12 Auer Cape\nRosamondstad, KS 92288",
    city: "New Sasha",
    country: "Jersey",
    phoneNumber: "299-484-8138",
    email: "Gutmann.Mohammed@yahoo.com",
    __typename: "User",
  },
  {
    uuid: "65a55f30-5003-42e1-a4f2-9dddcc5d8983",
    fullName: "Iva Treutel",
    address: "75 Kuphal Locks Suite 812\nWest Sage, CT 02997-1259",
    city: "East Rhianna",
    country: "Italy",
    phoneNumber: "765-489-1641",
    email: "Daugherty.Gladyce@Howe.biz",
    __typename: "User",
  },
  {
    uuid: "d3842d47-504a-4107-98b3-e8ed5959e4e5",
    fullName: "Amy Schinner",
    address: "8481 Ezra Track Apt. 220\nEast Brittany, CA 06194",
    city: "Florinechester",
    country: "Croatia",
    phoneNumber: "188-978-9135",
    email: "Shields_Carey@Malinda.biz",
    __typename: "User",
  },
  {
    uuid: "bef8a2ba-adab-488c-8bad-5a30427c7728",
    fullName: "Joyce Blick",
    address: "340 Bogan Overpass Apt. 014\nBlancabury, DE 22363",
    city: "Nitzschemouth",
    country: "Namibia",
    phoneNumber: "790-139-0749",
    email: "Kling_Hope@Weissnat.biz",
    __typename: "User",
  },
  {
    uuid: "903d7931-e08e-4b78-94cf-f290ca4e8c5c",
    fullName: "Fabian Fahey",
    address: "6404 Ruthie Rue\nSouth Reinholdbury, NM 25629",
    city: "North Rodrick",
    country: "Armenia",
    phoneNumber: "648-494-8008",
    email: "Mozelle_Beatty@Teagan.biz",
    __typename: "User",
  },
  {
    uuid: "5879ca64-c022-49a1-a718-0a65f8b3995d",
    fullName: "Alison Bahringer",
    address: "288 Langosh Glens Apt. 109\nLake Keaganhaven, IN 99002-9258",
    city: "East Ida",
    country: "Eritrea",
    phoneNumber: "528-766-8557",
    email: "Pagac.Brandyn@hotmail.com",
    __typename: "User",
  },
  {
    uuid: "79176b0b-82db-44c5-b78c-7125c4d57536",
    fullName: "Weldon Schmeler",
    address: "424 Ramona Gateway Apt. 089\nCummerataburgh, OR 81870-4532",
    city: "Remingtonside",
    country: "Timor-Leste",
    phoneNumber: "720-628-4332",
    email: "Cloyd.Aufderhar@yahoo.com",
    __typename: "User",
  },
  {
    uuid: "6d83515e-19e2-4fbf-a2e0-c01129935945",
    fullName: "Quincy Larson",
    address: "985 Kautzer Roads\nManuelaview, LA 75390-2593",
    city: "Kaleyport",
    country: "Saudi Arabia",
    phoneNumber: "092-732-6060",
    email: "Wunsch.Jennie@Narciso.io",
    __typename: "User",
  },
  {
    uuid: "c445e46d-f597-4893-b2a1-d2ae01e52ca1",
    fullName: "Hope Rosenbaum",
    address: "168 Fahey Walk Suite 182\nKubbury, VA 88185",
    city: "New Raven",
    country: "Netherlands",
    phoneNumber: "318-435-4596",
    email: "VonRueden_Lilyan@Magnus.us",
    __typename: "User",
  },
  {
    uuid: "b51c9177-5cfc-45a8-bf09-4a3e8ced8f4d",
    fullName: "Burnice Shanahan",
    address: "26 Ofelia Road Apt. 308\nNew Fern, OK 88034",
    city: "Ilenefurt",
    country: "Indonesia",
    phoneNumber: "151-777-6575",
    email: "Madyson_Hills@Mittie.name",
    __typename: "User",
  },
];
