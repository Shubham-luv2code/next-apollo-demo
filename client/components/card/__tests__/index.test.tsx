import React from "react";
import Card from "../index";
import { userMock } from "../../../__mocks__/userMock";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Card", () => {
  it("should match snapshot", () => {
    const component = renderer.create(<Card user={userMock[0]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should find the email address of user", () => {
    const { getByText } = render(<Card user={userMock[0]} />);
    const link = getByText("Carol_Zboncak@Trycia.co.uk");
    expect(link).toHaveAttribute("href", "Carol_Zboncak@Trycia.co.uk");
  });
});
