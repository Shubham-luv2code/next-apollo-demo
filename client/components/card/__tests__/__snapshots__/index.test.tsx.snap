// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div
  className="card grid-column"
>
  <div
    className="card-field"
  >
     
    <div
      className="user-name"
    >
      Stanton Goodwin
    </div>
  </div>
  <div
    className="card-field"
  >
     
    <div>
      586 Madonna Plain
West Christian, SC 90789, West Jaquelin, Ireland
    </div>
  </div>
  <div
    className="card-field"
  >
     
    <div>
      549-842-0265
    </div>
  </div>
  <div
    className="card-field"
  >
     
    <a
      href="Carol_Zboncak@Trycia.co.uk"
    >
      Carol_Zboncak@Trycia.co.uk
    </a>
  </div>
</div>
`;
